src/common/TheNavbar.vue

```diff
@@ -34,19 +48,34 @@ import { defineComponent } from "vue";
 
 import TheLogo from "@/common/TheLogo.vue";
 import AppIcon from "@/common/AppIcon.vue";
+import AppButton from "@/common/AppButton.vue";
 import { routes } from "../router";
 
+import userClient from "../user/client";
+
 export default defineComponent({
   name: "TheNavbar",
-  components: { AppIcon, TheLogo },
+  components: { AppIcon, TheLogo, AppButton },
   data() {
     return {
       isNavOpen: false
+      // isLoggedIn: true
     };
   },
   computed: {
     links() {
       return routes;
+    },
+    isLoggedIn() {
+      return userClient.currentUser() != null;
+    }
```

Comment:
Vue's state updates is a bit different than React in that it's implicit - you don't do `setState` anywhere when you're using Vue. Instead, Vue has this notion of "reactive objects". Whenever you access a property of a reactive object, Vue makes sure that property stays up to date. For example:

```js
data() {
  return { a: 5 } // this is a reactive object
},
computed: {
  b() {
    return this.a * 2 // this is accessing a property of a reactive object
  }
},
methods: {
  // this is updating the reactive object. When this runs, b gets recomputed because Vue 
  // knows b depends on a
  incrementA() {
    this.a++
  }
}
```

This means that for `b` to recompute, Vue has to _know_ that `b` depends on `a`. And the only way Vue can know that is if `a` is a reactive object. Generally speaking, anything from `data`, `computed`, or `props` is reactive.

The reason why `isLoggedIn` isn't getting updated is because you're not accessing any reactive objects inside the computed property. Unfortunately, there's not really a way to make external objects reactive, so you would have to do something like:

1. Put `isLoggedIn` into `data` and set it to some default value (probably `false`)
2. On mount, (1) update `isLoggedIn` based on if the user is logged in or not, and (2) attach an event listener to the firebase Auth app to listen to changes in login state
3. When the firebase login state changes, update `isLoggedIn` to whatever the new state is

Hope this clarifies a few things! @jamesmengo